src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -50,23 +39,154 @@ static ColliderCylinderInit D_80B18910 = {
     },
     { 20, 64, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B18360.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B183F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/EnTg_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/EnTg_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B185C0.s")
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, 0xFF };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/EnTg_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B18704.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B1871C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B18778.s")
+const ActorInit En_Tg_InitVars = {
+    ACTOR_EN_TG,
+    ACTORCAT_NPC,
+    FLAGS,
+    OBJECT_MU,
+    sizeof(EnTg),
+    (ActorFunc)EnTg_Init,
+    (ActorFunc)EnTg_Destroy,
+    (ActorFunc)EnTg_Update,
+    (ActorFunc)EnTg_Draw,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/EnTg_Draw.s")
+u16 func_80B18360(GlobalContext* globalCtx, Actor* thisx) {
+    EnTg* this = THIS;
+    u16 temp;
+    u32 phi;
+
+    temp = Text_GetFaceReaction(globalCtx, 0x24);
+    if (temp != 0) {
+        return temp;
+    }
+    if (globalCtx->sceneNum == 0x52) {
+        if (this->unk_208 & 1) {
+            phi = 0x5089;
+        } else {
+            phi = 0x508A;
+        }
+        return phi;
+    } else {
+        if (this->unk_208 & 1) {
+            phi = 0x7025;
+        } else {
+            phi = 0x7026;
+        }
+        return phi;
+    }
+}
+
+s16 func_80B183F8(GlobalContext* globalCtx, Actor* thisx) {
+    EnTg* this = THIS;
+
+    switch (func_8010BDBC(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 3:
+        case 4:
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+        case 9:
+            return 1;
+        case 2:
+            switch (this->actor.textId) {
+                case 0x5089:
+                case 0x508A:
+                    this->unk_208++;
+                    break;
+                case 0x7025:
+                case 0x7026:
+                    this->actor.params ^= 1;
+                    this->unk_208++;
+                    break;
+            }
+            return 0;
+        default:
+            return 1;
+    }
+}
+
+void EnTg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTg* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 28.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600AE40, &D_06005040, NULL, NULL, 0);
```

Comment:
```suggestion
    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_mu_Skel_00AE40, &object_mu_Anim_005040, NULL, NULL, 0);
```

Reply:
It would be ideal if you could check those animations and name them properly in the XML (`assets/xml/objects/object_mu.xml`). You should be able to see those animations in a tool like Z64Utils.
The naming scheme is: `g` ObjectName AnimationName `Anim`. For example: `gHoneyDarlingSpinAnim`.
If you need help with any of this feel free to ask here or in the server.

Also, naming those animations is not mandatory.

Reply:
Two things: firstly, this animation seems to solely be to prop the skeleton up (just like Deku Babas, all the interesting animation is manual 😆), so I don't think it needs a special name.

Secondly, and more importantly, these characters are *not* called Honey and Darling in OoT, just in MM. [The Encyclopedia](https://archive.org/details/TheLegendOfZeldaEncyclopedia/page/n229/mode/2up) calls them `DancingCouple`.

Reply:
In that case, the actor description at the top of this file should be updated accordingly.

Reply:
Good point.